hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/DefaultSparkRecordMerger.java

```diff
@@ -37,46 +35,38 @@
  */
 public class DefaultSparkRecordMerger extends HoodieSparkRecordMerger {
 
-  private String[] orderingFields;
-
   @Override
   public String getMergingStrategy() {
     return HoodieRecordMerger.EVENT_TIME_BASED_MERGE_STRATEGY_UUID;
   }
 
   @Override
-  public Option<Pair<HoodieRecord, Schema>> merge(HoodieRecord older, Schema oldSchema, HoodieRecord newer, Schema newSchema, TypedProperties props) throws IOException {
-    Option<Pair<HoodieRecord, Schema>> deleteHandlingResult = handleDeletes(older, oldSchema, newer, newSchema, props);
-    if (deleteHandlingResult != null) {
-      return deleteHandlingResult;
-    }
-
-    if (orderingFields == null) {
-      orderingFields = ConfigUtils.getOrderingFields(props);
-    }
-    if (older.getOrderingValue(oldSchema, props, orderingFields).compareTo(newer.getOrderingValue(newSchema, props, orderingFields)) > 0) {
-      return Option.of(Pair.of(older, oldSchema));
+  public <T> BufferedRecord<T> merge(BufferedRecord<T> older, BufferedRecord<T> newer, RecordContext<T> recordContext, TypedProperties props) throws IOException {
+    if (older.getOrderingValue().compareTo(newer.getOrderingValue()) > 0) {
```

Comment:
we may need to handle the special commit time ordering value `0` so that the `#compareTo` not throws when the ordering value is not a Long.

Reply:
Something like check if the records are commit time ordered deletes before this?